test/wasm_test.cc

```diff
@@ -113,4 +113,131 @@ TEST_P(TestVm, GetOrCreateThreadLocalWasmFailCallbacks) {
+    canary_count++;
+    return std::make_shared<WasmHandleBase>(wasm);
+  };
+
+  PluginHandleFactory plugin_handle_factory =
+      [](const std::shared_ptr<WasmHandleBase> &base_wasm,
+         const std::shared_ptr<PluginBase> &plugin) -> std::shared_ptr<PluginHandleBase> {
+    return std::make_shared<PluginHandleBase>(base_wasm, plugin);
+  };
+
+  // Read the minimal loadable binary.
+  auto source = readTestWasmFile("configure_check.wasm");
+
+  WasmHandleFactory wasm_handle_factory_baseline =
+      [this, vm_ids, vm_configs](std::string_view vm_key) -> std::shared_ptr<WasmHandleBase> {
+    auto base_wasm = std::make_shared<TestWasm>(newVm(), vm_ids[0], vm_configs[0], vm_key);
+    return std::make_shared<WasmHandleBase>(base_wasm);
+  };
+
+  // Create a baseline plugin.
+  const auto plugin_baseline = std::make_shared<PluginBase>(
+      plugin_name, root_ids[0], vm_ids[0], engine_, plugin_configs[0], fail_open, plugin_keys[0]);
+
+  const auto vm_key_baseline = makeVmKey(vm_ids[0], vm_configs[0], "common_code");
+  // Create a base Wasm by createWasm.
+  auto wasm_handle_baseline =
+      createWasm(vm_key_baseline, source, plugin_baseline, wasm_handle_factory_baseline,
+                 wasm_handle_clone_factory_for_canary, false);
+  ASSERT_TRUE(wasm_handle_baseline && wasm_handle_baseline->wasm());
+
+  // Check if it ran for baseline root context
+  EXPECT_TRUE(TestContext::isGlobalLogged("onConfigure: " + root_ids[0]));
+  // For each create Wasm, canary should be done.
+  EXPECT_EQ(canary_count, 1);
+
+  for (const auto &root_id : root_ids) {
+    for (const auto &vm_id : vm_ids) {
+      for (const auto &vm_config : vm_configs) {
+        for (const auto &plugin_key : plugin_keys) {
+          for (const auto &plugin_config : plugin_configs) {
+            canary_count = 0;
+            TestContext::resetGlobalLog();
+            WasmHandleFactory wasm_handle_factory_comp =
+                [this, vm_id,
+                 vm_config](std::string_view vm_key) -> std::shared_ptr<WasmHandleBase> {
+              auto base_wasm = std::make_shared<TestWasm>(newVm(), vm_id, vm_config, vm_key);
+              return std::make_shared<WasmHandleBase>(base_wasm);
+            };
+            const auto plugin_comp = std::make_shared<PluginBase>(
+                plugin_name, root_id, vm_id, engine_, plugin_config, fail_open, plugin_key);
+            const auto vm_key = makeVmKey(vm_id, vm_config, "common_code");
+            // Create a base Wasm by createWasm.
+            auto wasm_handle_comp =
+                createWasm(vm_key, source, plugin_comp, wasm_handle_factory_comp,
+                           wasm_handle_clone_factory_for_canary, false);
+
+            if (plugin_config.empty()) {
+              // configure_check.wasm should raise the error at `onConfigure` in canary when the
+              // `plugin_config` is empty string.
+              EXPECT_EQ(wasm_handle_comp, nullptr);
+              return;
```

Comment:
Should this be `continue` rather than `return`? ISTM `return` would cut short the entire test case.

Reply:
Oops.. Thanks for checking this!